packages/graphql/src/schema/make-augmented-schema.ts

```diff
@@ -133,6 +164,25 @@ function makeAugmentedSchema(
         (x) => x.kind === "InterfaceTypeDefinition"
     ) as InterfaceTypeDefinitionNode[];
 
+    if (interfaces.length) {
+        if (interfaces.find((x) => x.name.value === "Node")) {
+            throw new Error(
+                "Interface name `Node` reserved to support the node interface. See https://relay.dev/docs/guides/graphql-server-specification"
+            );
+        }
+    }
+
+    const nodeTypeDef = `
```

Comment:
Ok that makes sense. I can see how it would be tricky to migrate an existing data set. 

One thing to note is that the Relay specification has a few steadfast rules, and having an id property on all implementing nodes [is required by the specification](https://graphql.org/learn/global-object-identification/#introspection). 

One thought I had was to make the changes by this PR (i.e. full relay implementation) optional that you could set with a flag in the config. I believe I recall seeing that in one of the Relationship-Properties proposals.

Reply:
So I think the solution to this is to use id encoding. All we would need from users is an identification of a unique field property. Then you can encode the TypeName and the uniqueValue into a string that you use for the id. This way the id is globally unique amongst types, you can auto-generate it based on identified unique keys, etc.

For example, using the [graphql-relay-js](https://github.com/graphql/graphql-relay-js/blob/fa978fbfaa367b79b4616d091471be87e84f3dbe/src/node/node.js#L87) library again as example, there are two helpers: 

```Typescript
/**
 * Takes a type name and an ID specific to that type name, and returns a
 * "global ID" that is unique among all types.
 */
export function toGlobalId(type: string, id: string | number): string {
  return base64([type, GraphQLID.serialize(id)].join(':'));
}

/**
 * Takes the "global ID" created by toGlobalID, and returns the type name and ID
 * used to create it.
 */
export function fromGlobalId(globalId: string): ResolvedGlobalId {
  const unbasedGlobalId = unbase64(globalId);
  const delimiterPos = unbasedGlobalId.indexOf(':');
  return {
    type: unbasedGlobalId.substring(0, delimiterPos),
    id: unbasedGlobalId.substring(delimiterPos + 1),
  };
}
```

So long as the id value is unique across the Node, by encoding it with the typeName, it will be unique across all types.

For users that don't have an id field, or want to use another field, you can autogenerate it for them. All you'll need them to do is identify the field with the @id directive?

```graphql
# using uuid
type Movie {
  id: ID!

  _id: ID @id
}

# using other field
type Actor {
  id: ID!
 
  imdbNo: String! @id(autogenerate: false) 
}
``` 

This would also make node resolution easier since all you'll have to do is decode the id, so two database trips aren't required.